universe/ScriptingContext.h

```diff
@@ -9,6 +9,20 @@
 
 class UniverseObject;
 
+struct BackgroundContext {
```

Comment:
why "Background" ?

Reply:
Couldnt find a good name. We have the context info which changes while the conditions get evaluated. The background is what does not change. Maybe i meant rather backdrop(?).

Reply:
Not sure what you mean by the scripting context changing "while the conditions get evaluated"... how so? It's passed by const reference to condition classes `Eval` functions. It's sometimes used to create a new scripting context for sub-evaluations of conditions or value refs, but the original doesn't change. I don't think this needs a separate struct.

Reply:
If I got you right you said the same visibility map gets handed through to the sub-evals. The same would be true for a bout value. The idea is that i can put all that data that stays the same in the same data structure and use a single reference to that.

So I should have said "During the evaluation of a condition/valueRef tree there are different values for e.g. localCandidate necessary while some values are always the same."

With reference to vis map and int val in ScriptingContext: every time a ScriptingContext for a subevaluation gets created, both get copied (with the old implementation i think even the whole map got copied every time)

With a reference to a const data structure in ScriptingContext where the data structure contains bout and map : every time a ScriptingContext for a subevaluation gets created, only one reference gets copied. There is are more addresses to resolve, but only once for each Visibility condition or CombatBout condition which is neglible. 

While the reasoning for two values this might be overengineering it is easy now to add more values cost efficiently if we need to.

Reply:
Allocating and copying one int and one reference isn't enough to warrant wrapping in a separate struct yet, I think. If a bunch more gets added, then perhaps in future it would be worthwhile, especially in the wrapped up stuff is conceptually related.